adapter/internal/eventhub/subscription.go

```diff
@@ -116,6 +119,32 @@ func LoadSubscriptionData(configFile *config.Config) {
 	var responseChannel = make(chan response)
 	for _, url := range resources {
 		go InvokeService(url.endpoint, url.responseType, nil, responseChannel, 0)
+		for {
+			data := <-responseChannel
+			logger.LoggerSync.Debug("Receiving data for an environment")
+			if data.Payload != nil {
+				logger.LoggerSync.Debug("Payload data recieved")
+				retrieveSubscriptionDataFromChannel(data)
+				break
+			} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
+				logger.LoggerSync.Errorf("Error occurred when retrieving Subscription information from the control plane: %v", data.Error)
+				health.SetControlPlaneRestAPIStatus(false)
+			} else {
+				// Keep the iteration going on until a response is recieved.
+				logger.LoggerSync.Errorf("Error occurred while fetching data from control plane: %v", data.Error)
+				go func(d response) {
+					// Retry fetching from control plane after a configured time interval
+					if conf.ControlPlane.RetryInterval == 0 {
+						// Assign default retry interval
+						conf.ControlPlane.RetryInterval = 5
```

Comment:
Can't we assign the default value in the struct

Reply:
Is this a constant interval?

Reply:
It is 5 by default in the default config, but I think it is safe to have this here. This was done in a similar way for the api fetcher retry code as well.

Reply:
we could change in both places I guess. Because this API fetch retrying is implemented way before the default value implementation.